### PR TITLE
[all] Convert some loops to range-based

### PR DESCRIPTION
**Pull request by alxbilger**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/STLExporter.cpp [hunk 1]
```diff
@@ -137,24 +137,24 @@ bool STLExporter::writeSTL(bool autonumbering)
     }
     if(!triangleIndices.empty())
     {
-        for(unsigned int i=0;i<triangleIndices.size();i++)
+        for(auto triangleIndice : triangleIndices)
```
Comment (fredroy) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/STLExporter.cpp]:
```suggestion
        for(const auto& triangleIndex : triangleIndices)
```

const ref should be usable

(this applies to the other changes in STLExporter, lazy to do the other suggestions 😅)

File: Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedProjectiveConstraint.cpp [hunk 2]
```diff
@@ -89,8 +89,8 @@ void FixedProjectiveConstraint<Rigid3Types>::draw(const core::visual::VisualPara
         }
         else
         {
-            for (SetIndex::const_iterator it = indices.begin(); it != indices.end(); ++it)
-                points.push_back(x[*it].getCenter());
+            for (const unsigned int indice : indices)
+                points.push_back(x[indice].getCenter());
```
Comment (fredroy) [inline Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedProjectiveConstraint.cpp]:
```suggestion
            for (const unsigned int index: indices)
                points.push_back(x[index].getCenter());
```

I guess clang-tidy does not know that the singular of indices is index 😅

**Review by fredroy — CHANGES_REQUESTED** [review]
The "indice" thing to replace with "index" appears multiple times (I just highlighted one...)

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/STLExporter.cpp (hunk 1 above)
Comment (alxbilger) [reply Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/STLExporter.cpp]: if I am not wrong, `triangleIndice` is a primitive type so it's preferable to pass by copy, no?

Comment (alxbilger) [reply Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/STLExporter.cpp]: ah no, my bad, it's a vector of triangle, not an array of indices.

**Comment by bakpaul** [comment]
[ci-build][with-all-tests][force-full-build]

**Comment by bakpaul** [comment]
[ci-build][with-all-tests]

**Review by fredroy — CHANGES_REQUESTED** [review]
UniformMass still uses `indice` 😅

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/STLExporter.cpp [hunk 3]
```diff
@@ -137,24 +137,24 @@ bool STLExporter::writeSTL(bool autonumbering)
...
         {
-            vecTri.push_back(triangleIndices[i]);
+            vecTri.push_back(triangleIndex);
         }
     }
     else if(!quadIndices.empty())
     {
         BaseMeshTopology::Triangle tri;
-        for(unsigned int i=0;i<quadIndices.size();i++)
+        for(const auto& quadIndice : quadIndices)
```
Comment (fredroy) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/STLExporter.cpp]:
```suggestion
        for(const auto& quadIndex : quadIndices)
```
and others in this file 🙊

**Review by fredroy — CHANGES_REQUESTED** [review]

**Review by fredroy — APPROVED** [review]

**Pull request merged**